hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java

```diff
@@ -264,10 +264,9 @@ private CountryAuthnRequestGeneratorService getCountryAuthnRequestGeneratorServi
                                                                                        Function<EidasAuthnRequestFromHub, String> eidasRequestStringTransformer,
                                                                                        EidasAuthnRequestTranslator eidasAuthnRequestTranslator,
                                                                                        @Named("HubEidasEntityId") Optional<String> hubEidasEntityId) {
-        if (!hubEidasEntityId.isPresent()) {
-            throw new InvalidConfigurationException(EIDAS_HUB_ENTITY_ID_NOT_CONFIGURED_ERROR_MESSAGE);
-        }
-        return new CountryAuthnRequestGeneratorService(countrySingleSignOnServiceHelper.get(), eidasRequestStringTransformer, eidasAuthnRequestTranslator, hubEidasEntityId.get());
+        return new CountryAuthnRequestGeneratorService(
+                countrySingleSignOnServiceHelper.orElseThrow(() -> new InvalidConfigurationException(EIDAS_HUB_ENTITY_ID_NOT_CONFIGURED_ERROR_MESSAGE)),
+                eidasRequestStringTransformer, eidasAuthnRequestTranslator, hubEidasEntityId.orElseThrow());
```

Comment:
Oof, I think this one got a bit unwieldy. It's changed the logic, so that it no longer throws based on `hubEidasEntityId` being empty, rather it throws on `countrySingleSignOnServiceHelper` being empty.

suggested change, which keeps the logic the same (for better or worse) and makes it a bit more readable.
```
return hubEidasEntityId.map(eid -> new CountryAuthnRequestGeneratorService(
                countrySingleSignOnServiceHelper.get(),
                eidasRequestStringTransformer,
                eidasAuthnRequestTranslator, eid))
                .orElseThrow(() -> new InvalidConfigurationException(EIDAS_HUB_ENTITY_ID_NOT_CONFIGURED_ERROR_MESSAGE));
```

Reply:
ugh. That'll teach me to pay closer attention